jdaviz/app.py

```diff
@@ -95,13 +95,17 @@ def to_unit(self, data, cid, values, original_units, target_units):
         # gives the units of the values array, which might not be the same
         # as the original native units of the component in the data.
         if cid.label == "flux":
-            spec = data.get_object(cls=Spectrum1D)
-            if len(values) == 2:
-                # Need this for setting the y-limits
-                spec_limits = [spec.spectral_axis[0].value, spec.spectral_axis[-1].value]
-                eqv = u.spectral_density(spec_limits*spec.spectral_axis.unit)
+            try:
+                spec = data.get_object(cls=Spectrum1D)
+            except Exception:
```

Comment:
is the bare exception here safe or could it get us until trouble in any way?  Can we do something to check in advance instead or check for a specific expected exception?

Reply:
The original traceback is very confusing so I don't know what is the exact exception to catch here. In general, I think falling back to not using spectral axis should be okay when Spectrum1D conversion fails?

Reply:
so the consequence of the exception being triggered (and returning an empty list) is just that that data entry will not show any valid units to convert?

Reply:
Let's ask @javerbukh since I am not very sure where the output goes. Either that or it only uses basic equivalencies that might succeed or not.

Reply:
This is used when data with a different spectral axis unit from the display data is visualized simultaneously with the display data in a viewer. For example data 1 with `um` is loaded and data 2 with `AA` is added, `to_unit` will convert the data from data 2 so that it matches the units of data 1. An empty list may result in the data not being visualized at all, so we should check that that actually happens and add a snackbar or other message so the user understands why the data was not added.

That said, if `spec = data.get_object(cls=Spectrum1D)` is throwing an error, something else seems to be broken and this should just temporarily get us around that issue.

TLDR: This looks like a good temporary fix but we should track the larger issue of why this is throwing an exception.

Reply:
I'm just worried that any future bug within `get_object` will be very hard to track down with this blanket exception 🤔

Reply:
Should I close this without merge then, and someone else can pick it up next sprint for a proper fix?

Reply:
seems to be `RuntimeError`... but I'll look into if we can feasilby predict in advance instead of relying on try/except